pageserver/src/walingest.rs

```diff
@@ -503,6 +524,24 @@ impl WalIngest {
         Ok(modification.len() > prev_len)
     }
 
+    /// This is the same as AdjustToFullTransactionId(xid) in PostgreSQL
+    fn adjust_to_full_transaction_id(&self, xid: TransactionId) -> Result<u64> {
+        let next_full_xid = self.checkpoint.nextXid.value;
+
+        let next_xid = (next_full_xid) as u32;
+        let mut epoch = (next_full_xid >> 32) as u32;
+
+        if xid > next_xid {
+            // Wraparound occurred, must be from a prev epoch.
+            if epoch == 0 {
+                bail!("apparent XID wraparound with prepared transaction XID {xid}, nextXid is {next_full_xid}");
+            }
+            epoch -= 1;
```

Comment:
May be just use `checked_sub` instead if explicit check?

Reply:
Perhaps, but I preferred to keep it more similar to the corresponding postgres function, which does:
```
	if (unlikely(xid > nextXid))
	{
		/* Wraparound occurred, must be from a prev epoch. */
		Assert(epoch > 0);
		epoch--;
	}
```
it's not quite the same anyway, with the Assert, but still..